react/src/js/react/posts/PostView.jsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import Reflux from "reflux";
+
+import { PostsStore, PostsActions } from "../discover/PostsStore";
+import LoadingComponent from "../misc/LoadingComponent";
+import Post from "./Post";
+
+export default class PostView extends Reflux.Component {
+    constructor(props) {
+        super(props);
+        this.store = PostsStore;
+    }
+
+    componentDidMount() {
+        PostsActions.getPost(this.props.match.params.id);
+    }
+
+    render() {
+        if (this.state.currentPost.length === 0) {
```

Comment:
This function should render an error if one occurred while fetching the post. This if statement should probably be: if (this.state.fetchingPost) { return Loader } else if (error loading post) { return Alert }

Reply:
Since I'm fetching the post in the componentDidMount, that state isn't set yet...... should be fetching somewhere else and then checking the state?